src/components/progress-circle/progress-circle.ts

```diff
@@ -41,47 +84,172 @@ export class MdProgressCircle {
    * mode is bound to the host as the attribute host.
    */
   @HostBinding('attr.mode')
-  @Input() mode: 'determinate' | 'indeterminate' = 'determinate';
+  @Input()
+  get mode() {
+    return this._mode;
+  }
+  set mode(m: 'determinate' | 'indeterminate') {
```

Comment:
Rather than recreating the string literal type each time, you can declare it once and reuse it:
```ts
export type ProgressCircleMode = 'determinate' | 'indeterminate';
```